Module4_Labs/Lab2_Doubly_Linked_List/Cards/4.md

```diff
@@ -1,12 +1,49 @@
-<!--title={The Size of the Linked List}--> 
+<!--title={Insert an Item Into Doubly Linked List}--> 
 
-<!--badges={Algorithms:15,Python:15}-->
+<!--badges={Algorithms:30,Python:20}-->
 
-<!--concepts={Size of a Linked List}-->
+<!--concepts={Insert Into a Linked List}-->
 
-Although the keyword `list` in python has a built-in function called `len()` that return the number of entries in a list, we need to write a function to do the same thing for our doubly linked list.
+ ![Image result for cutting line people](https://philosophyviaphotos.files.wordpress.com/2018/11/source-ozfm9469_ext_20_en_34.jpg?w=490&h=350&crop=1)  
 
-Define a function called **getSize()** that returns the number of nodes in your doubly linked list.
+We will have the following five methods of inserting an item in doubly linked list: 
 
+- insert in an empty list
+- insert at the start
+- insert at the end
+- insert after another item
+- insert before another item
 
+These are extremely useful for editing a doubly linked list, because you can add in very specific ways with different methods.
 
+For each method, you will need to write a function that does the specific insertion in the doubly linked list. Remember that all manipulations to the linked list require careful modification of the pointers. A useful tip would be writing a doubly linked list on a piece of paper with proper pointers pointing at specific nodes. As you design the functions, you can erase the original pointing line and draw a new one to see what happens.
+
+Below are the descriptions of each method that you will implement:
+
+**Inserting Items in Empty List**
+
+- Write a function called **push()** that takes in a element as the parameter, and check whether the list is empty.
+- If this is an empty list, insert the new node. Otherwise, you will continue to the next step below, "inserting items at the start."
+
+**Inserting Items at the Start**
```

Comment:
if the methods have parameter, please include them in the method headers that are introduced here.